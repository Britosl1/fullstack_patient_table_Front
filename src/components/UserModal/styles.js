import styled from 'styled-components';

export const ModalTitle = styled.p`
  font-size: 30px;
  color: ${props => props.color};
  font-weight: 600;
`

export const ModalImageContainer = styled.div`
  display: flex;
  justify-content: center;
  margin: 30px;
`

export const ModalBody = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 0px 20px;
`

export const ModalLine = styled.div`
  border: solid 2px;
  border-color: ${props => props.color};
  width: 100%;
  margin: 8px 0px;
`

export const ModalImage = styled.img`
  border-radius: 50px;
`
export const ModalInfo = styled.div`
  color: ${props => props.color};
  display: flex;
  /* justify-content: space-between; */
  font-size: 15px;
  width: 70%;
`
export const ModalInsideText = styled.text`
  font-size: 18px;
  color: ${props => props.color};
  font-weight: 300px;
`

export const ButtonContainer = styled.div`
  margin: 0px 10px;
`