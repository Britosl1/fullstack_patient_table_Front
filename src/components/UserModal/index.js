import React from 'react';
import { Modal, Button } from 'react-bootstrap'
import axios from 'axios';
import { url } from '../../service/api/index';
import { ModalTitle, ModalInfo, ModalInsideText, ModalImage, ModalImageContainer, ModalLine, ModalBody, ButtonContainer } from './styles';
import { tableColors } from '../../service/styles/colors';

function UserModal({
  show,
  handleClose,
  image,
  name,
  email,
  nat,
  gender,
  dob,
  phone,
  location,
}) {

  const updateUser = async (id) => {
    try {
      const response = await axios.put(`${url}/${id}`);
      const data = response.data;

      const results = data;

      console.log(results);
      return results;
    } catch (e) {
      console.log(e);
    }
  };

  const deleteUser = async (id) => {
    try {
      const response = await axios.delete(`${url}/${id}`);

      return response;
    } catch (e) {
      console.log(e);
    }
  };

  return (
    <Modal show={show} onHide={handleClose} >
      <Modal.Header closeButton>
        <ModalTitle color={tableColors.burntSienna}>User Info</ModalTitle>
      </Modal.Header>
      <ModalImageContainer>
        <ModalImage src={image}></ModalImage>
      </ModalImageContainer>
      <ModalBody>
        <ModalLine color={tableColors.charcoal} />
        <ModalInfo>
          <ModalInsideText color={tableColors.burntSienna}>Email:&nbsp;</ModalInsideText>
          <ModalInsideText >{email}</ModalInsideText>
        </ModalInfo>
        <ModalInfo>
          <ModalInsideText color={tableColors.burntSienna}>Name:&nbsp;</ModalInsideText>
          <ModalInsideText >{name}</ModalInsideText>
        </ModalInfo>
        <ModalInfo>
          <ModalInsideText color={tableColors.burntSienna}>Nationality:&nbsp;</ModalInsideText>
          <ModalInsideText >{nat}</ModalInsideText>
        </ModalInfo>
        <ModalInfo>
          <ModalInsideText color={tableColors.burntSienna}>Gender:&nbsp;</ModalInsideText>
          <ModalInsideText >{gender}</ModalInsideText>
        </ModalInfo>
        <ModalInfo>
          <ModalInsideText color={tableColors.burntSienna}>Age:&nbsp;</ModalInsideText>
          <ModalInsideText >{dob}</ModalInsideText>
        </ModalInfo>
        <ModalInfo>
          <ModalInsideText color={tableColors.burntSienna}>Phone:&nbsp;</ModalInsideText>
          <ModalInsideText >{phone}</ModalInsideText>
        </ModalInfo>
        <ModalInfo>
          <ModalInsideText color={tableColors.burntSienna}>Location:&nbsp;</ModalInsideText>
          <ModalInsideText >{location}</ModalInsideText>
        </ModalInfo>
      </ModalBody>
      <Modal.Footer>
        <ButtonContainer>
          <Button onClick={updateUser}>
            Update
          </Button>
        </ButtonContainer>
        <Button variant="dark" onClick={deleteUser}>
          Delete
        </Button>
      </Modal.Footer>
    </Modal>
  )
}

export default UserModal;