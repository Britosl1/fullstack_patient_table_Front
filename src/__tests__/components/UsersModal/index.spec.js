import React from 'react';
import renderer from 'react-test-renderer';
import UserModal from '../../../components/UserModal/index'

describe('Snapshot test', () => {
  it('renders correctly', () => {

    const nav = renderer
      .create(<UserModal />).toJSON();

    expect(nav).toMatchSnapshot();
  });
});
