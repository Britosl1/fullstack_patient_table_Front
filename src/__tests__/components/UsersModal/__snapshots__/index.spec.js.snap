// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders correctly 1`] = `null`;
